stacks-signer/src/config.rs

```diff
@@ -338,19 +338,23 @@ pub fn build_signer_config_tomls(
     timeout: Option<Duration>,
     network: &Network,
     password: &str,
+    mut port_start: usize,
 ) -> Vec<String> {
     let mut signer_config_tomls = vec![];
 
-    let mut port = 30000;
-    let run_stamp = rand::random::<u16>();
-    let db_dir = format!(
-        "/tmp/stacks-node-tests/integrations-signers/{:#X}",
-        run_stamp,
-    );
-    fs::create_dir_all(&db_dir).unwrap();
-    for (ix, stacks_private_key) in stacks_private_keys.iter().enumerate() {
-        let endpoint = format!("localhost:{}", port);
-        port += 1;
+    for stacks_private_key in stacks_private_keys {
+        let endpoint = format!("localhost:{}", port_start);
+        port_start += 1;
+
+        let stacks_public_key = StacksPublicKey::from_private(stacks_private_key).to_hex();
+        let db_dir = std::env::temp_dir().join(format!(
+            "stacks-node-tests/integrations-signers/signer_{stacks_public_key}"
+        ));
```

Comment:
I'd prefer specifically using `/tmp/stacks-node/tests` rather than `temp_dir()` -- its what the rest of the integration tests do, and it makes it much easier to have them always in the same place as the rest of the test directories.

This would also mean that the run stamp should be added back in.

Reply:
Sure thing, I'd prefer to use `temp_dir()` instead of hard-coding `/tmp` but I see it's hard-coded in more places so that's a separate refactor.

Why should the run stamp be added back in? It was randomly generated within the function so it does not seem to correspond to any other directory. Also, I can't have this function be non-deterministic because I need a restarted signer to be able to read it's signerdb - which is why I'm using the public key to differentiate them here instead of a random run stamp.

Reply:
Done :heavy_check_mark: